src/solace_agent_mesh/common/utils/message_utils.py

```diff
@@ -0,0 +1,84 @@
+"""Message utility functions for size calculation and validation.
+
+This module provides utilities for calculating and validating message sizes
+to ensure they don't exceed configured limits. The size calculation matches
+the exact serialization format used by the Solace AI Connector (JSON + UTF-8).
+"""
+
+import json
+from typing import Any, Dict, Tuple
+
+from solace_ai_connector.common.log import log
+
+# Maximum bytes per character in UTF-8 encoding (4 bytes for full Unicode range)
+MAX_UTF8_BYTES_PER_CHARACTER = 4
+
+
+def calculate_message_size(payload: Dict[str, Any]) -> int:
+    """Calculate the exact size of a message payload in bytes.
+
+    Uses JSON serialization followed by UTF-8 encoding to match the exact
+    format used by the Solace AI Connector.
+
+    Args:
+        payload: The message payload dictionary to calculate size for.
+
+    Returns:
+        The size of the payload in bytes.
+
+    Note:
+        If JSON serialization fails, falls back to string representation
+        for size estimation.
+    """
+    try:
+        # Use JSON serialization + UTF-8 encoding to match Solace AI Connector
+        json_str = json.dumps(payload, separators=(",", ":"), ensure_ascii=False)
```

Comment:
I am a bit concerned about how expensive this might be. I know that if we don't do this, it would be impossible to get an accurate size, but I wonder if we really need that level of accuracy. 

For now, we should probably just go ahead with it since this optimization is not our primary concern. Perhaps in the future if we are needing to streamline performance, we will have to do it differently.